include/deal.II/grid/tria_faces.h

```diff
@@ -71,6 +72,13 @@ namespace internal
        */
       std::vector<unsigned char> quads_line_orientations;
 
+      /**
+       * Geometric entity type of each quad.
+       *
+       * @note Used only for dim=3.
+       */
+      std::vector<GeometricEntityType> quad_entity_type;
```

Comment:
Why would these not always have the same value, namely `QUAD`?

Reply:
Good point: the name is chosen a bit confusing. I guess we should rename all `quad_*` at some time because they can be also triangles. Any suggestions how to abstract quadrilateral and triangles?

Reply:
Let's keep it like this for now. PR #10631 will fill this array also with `ReferenceCell::Types::Tri` for tets/wedges/pyramids.

Making the naming of 2 entities (quad/tri) consistent is larger clean up effort (see e.g.: `GeometryInfo::quads_per_cell`).